### PR TITLE
CI improvements

### PR DESCRIPTION
**Pull request by thibault-martinez**

### CODE REVIEW

File: .github/workflows/bindings-java.yml [hunk 1]
```diff
@@ -1,34 +1,28 @@
...
     paths:
-      - ".github/workflows/examine-java-bindings.yml"
+      - ".github/workflows/bindings-java.yml"
       - ".github/actions/**"
       - "client/bindings/java/**"
-      # Watch the Rust core too
-      - "client/src/**"
-      - "client/Cargo.toml"
-      - "Cargo.lock"
+      - "**.rs"
```
Comment (Thoralf-M) [inline .github/workflows/bindings-java.yml]: With this is would also run if we change something in the Rust part of nodejs, no?

Comment (thibault-martinez) [reply .github/workflows/bindings-java.yml]: Ah yeah right. Do you think it's better to list all relevant rust file paths or to have `**.rs` and exclude what is not relevant? I fear that with listing all relevant, we end up forgetting to add new code. Like we forgot to add `types`.

Comment (Thoralf-M) [reply .github/workflows/bindings-java.yml]: Exclude sounds like the safer way

File: .github/workflows/bindings-java.yml [hunk 2]
```diff
@@ -54,7 +48,7 @@ jobs:
           cache: true
           cache-root: client/bindings/java/lib/native/
           cache-job-id: ${{ github.workflow }}-${{ github.job }}-${{ matrix.os }}-${{ matrix.java }}
```
Comment (Thoralf-M) [inline .github/workflows/bindings-java.yml:56]: Does it make sense to remove the language specific caches now? `-${{ matrix.java }}`

Comment (thibault-martinez) [reply .github/workflows/bindings-java.yml:56]: Yeah maybe

Comment (thibault-martinez) [reply .github/workflows/bindings-java.yml:56]: But if you allow me I'll do a dedicated PR for caching, there are too many things to explore and I need to chat with Rajiv.

File: .github/workflows/bindings-java.yml (hunk 1 above)
Comment (thibault-martinez) [reply .github/workflows/bindings-java.yml]: Done

**Review by Thoralf-M — APPROVED** [review]

**Review by kwek20 — APPROVED** [review]

**Pull request merged**